src/App.jsx

```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, sortField) {
+  const prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField.field) {
+        case SORT_FIELD.BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD.BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return sortField.isReversed
+    ? prepareGoods.reverse()
+    : prepareGoods;
+}
+
+const SORT_FIELD = {
+  BY_ALPHABET: 'alphabet',
+  BY_LENGTH: 'length',
+};
+
+const defaultSortField = {
+  field: '',
+  isReversed: false,
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState(defaultSortField);
+
+  const visibleGoods
+    = getPreparedGoods(goodsFromServer, sortField);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const { field, isReversed } = sortField;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': !(field === SORT_FIELD.BY_ALPHABET) })}
+          onClick={() => setSortField({
+            ...sortField,
+            field: SORT_FIELD.BY_ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': !(field === SORT_FIELD.BY_LENGTH) })}
+          onClick={() => setSortField({
+            ...sortField,
+            field: SORT_FIELD.BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button is-warning', {
+              'is-light': !isReversed,
+            })
+          }
+          onClick={() => setSortField({
+            ...sortField,
+            isReversed: !isReversed,
+          })}
+        >
+          Reverse
+        </button>
+
+        {field || isReversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => setSortField('')}
```

Comment:
The initial state of sortField is an object but here you are trying to set it to an empty string. This could lead to potential bugs in the future. It would be better to reset to the initial state object.